.github/workflows/weekly-cea.yml

```diff
@@ -0,0 +1,49 @@
+name: Weekly CEA CI
+
+on:
+  schedule:
+    - cron: "0 2 * * 6" # every Saturday at 2am UTC
+  workflow_dispatch:
+
+jobs:
+  build_and_test:
+    strategy:
+      matrix:
+        build_type:
+          - Release
+          - Debug
+        backend:
+          - name: cuda-a100
+            flags: -DKokkos_ENABLE_CUDA=ON -DKokkos_ARCH_AMPERE80=ON
+            modules: gcc/11.2.0/gcc-4.8.5 cuda/12.2.1/gcc-11.2.0 cmake/3.28.3/gcc-11.2.0
+            queue: gpua100
+            runner: [self-hosted, cuda]
+
+    runs-on: ${{ matrix.backend.runner }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Configure
+        run: |
+          module load ${{ matrix.backend.modules }}
+          cmake -B build \
+            -DCMAKE_BUILD_TYPE=${{ matrix.build_type }} \
+            -DCMAKE_CXX_STANDARD=17 \
+            -DCMAKE_CXX_FLAGS=-Werror \
+            -DKokkos_ENABLE_COMPILER_WARNINGS=ON \
+            -DKokkos_ENABLE_TESTS=ON \
+            -DKokkos_ENABLE_EXAMPLES=ON \
+            ${{ matrix.backend.flags }}
+
+      - name: Build
+        run: |
+          module load ${{ matrix.backend.modules }}
```

Comment:
Are you sure you need to load the modules at each job step ? I think you run them all in the same process, so I guess you could load them once in a dedicated step before the `Configure` step. Module loading failures would be easily identified then.

Reply:
Job steps are executed on the same runner, but [on different processes](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idsteps), so I cannot load modules for once. It's still possible to set environment variables from one step to the other, but [the process is tedious and has to be done for each of them](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#setting-an-environment-variable).